lib/reservation_mngr.rb

```diff
@@ -0,0 +1,78 @@
+require 'time'
+require 'date'
+
+
+require_relative 'reservation'
+
+module Hotel
+
+  class Reservation_mngr < Reservation
+
+    attr_reader :rooms, :reservations
+    attr_accessor :current_res_id
+
+    def initialize
+      @rooms = build_room_list
+      @reservations = []
+      @current_res_id = 1
+    end
+
+    def build_room_list
+      rooms = []
+      20.times do |i|
+        rooms << Hotel::Room.new({room_number: "#{i + 1}".to_i, price: 200})
+      end
+      return rooms
+    end
+
+    def check_dates(check_in, check_out)
+      raise ArgumentError if check_out == nil
+      raise ArgumentError if check_in == nil
+      raise ArgumentError if check_out < check_in
+      raise ArgumentError if check_in > check_out
+    end
+
+    def find_room(check_in, check_out)
+
+      @rooms.each do |room|
+        rm_avail = true
+        @reservations.each do |reservation|
+          if room.room_number == reservation.room.room_number
+            if Date.strptime(check_in, '%m/%d/%Y') >= reservation.check_in && Date.strptime(check_out, '%m/%d/%Y') < reservation.check_out
```

Comment:
I think that this conditional doesn't quite work. For example, assume the following things:
`room` is this:
```
#<Hotel::Room:0x00007f94d00d2728 @price=200, @room_number=1>
```
`@reservations` is this:
```
[#<Hotel::Reservation:0x00007f94d011bec8
  @check_in=#<Date: 2018-01-03 ((2458122j,0s,0n),+0s,2299161j)>,
  @check_out=#<Date: 2018-01-10 ((2458129j,0s,0n),+0s,2299161j)>,
  @cost=1400,
  @reservation_id=1,
  @room=#<Hotel::Room:0x00007f94d00d2728 @price=200, @room_number=1>>]
```
`reservation` is looking at the only entry in `@reservations`, so it is this:
```
#<Hotel::Reservation:0x00007f94d011bec8
 @check_in=#<Date: 2018-01-03 ((2458122j,0s,0n),+0s,2299161j)>,
 @check_out=#<Date: 2018-01-10 ((2458129j,0s,0n),+0s,2299161j)>,
 @cost=1400,
 @reservation_id=1,
 @room=#<Hotel::Room:0x00007f94d00d2728 @price=200, @room_number=1>>
```
the value of `check_in` is `"01/03/2018"`
the value of `check_out` is `"01/10/2018"`

...

Then the dates (check_in: 01/03/2018, checkout: 01/10/2018) and the reservation we're looking at (with check_in: 01/03/2018, checkout: 01/10/2018) should say that this room (room with id 1) is NOT available, but this conditional will evaluate as `false`, and therefore leave it `rm_avail` is `true` (and not go into the next line). This method will return the room with id 1 as the first available room.